File: lab5/bsqrt.c
float bsqrt(float x, float tol);
float abs(float x);

float bsqrt(float x, float tol) {
        float sqrt_x;
        sqrt_x = sqrt_x/2;
        while (abs(x-sqrt_x*sqrt_x) > tol) {
            x_sqrt = (x_sqrt + x/sqrt_x)/2;
        }
        return x_sqrt;
}

float abs(float x) {
        if (x>0)
                return x;
        else if (x<0)
                return -x;
        else
                return 0;
}

File: ex/bsqrt.py
def bsqrt(sqr, acc):
    est = sqr/2
    while abs(sqr,est) > acc:
        est = (est + sqr/est)/2
        print(est)
    return est

def abs(sqr1, est1):
	if est1*est1-sqr1 < 0:
		x = -1*(est1*est1-sqr1)
	else:
		x = est1*est1-sqr1
	return x

print(bsqrt(0, 0.000000001))
